plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/DefaultJdbcMetadata.java

```diff
@@ -284,32 +309,20 @@ private JdbcTableHandle flushAttributesAsQuery(ConnectorSession session, JdbcTab
             expressions.put(columnName, expression.get().getExpression());
         }
 
-        List<List<JdbcColumnHandle>> groupingSetsAsJdbcColumnHandles = groupingSets.stream()
-                .map(groupingSet -> groupingSet.stream()
-                        .map(JdbcColumnHandle.class::cast)
-                        .collect(toImmutableList()))
-                .collect(toImmutableList());
-
         List<JdbcColumnHandle> newColumnsList = newColumns.build();
 
         PreparedQuery preparedQuery = jdbcClient.prepareQuery(
                 session,
                 handle,
                 Optional.of(groupingSetsAsJdbcColumnHandles),
-                ImmutableList.<JdbcColumnHandle>builder()
-                        .addAll(groupingSetsAsJdbcColumnHandles.stream()
-                                .flatMap(List::stream)
-                                .distinct()
-                                .iterator())
-                        .addAll(newColumnsList)
-                        .build(),
+                newColumnsList,
                 expressions.build());
         handle = new JdbcTableHandle(
                 new JdbcQueryRelationHandle(preparedQuery),
                 TupleDomain.all(),
                 Optional.empty(),
                 OptionalLong.empty(),
-                Optional.of(newColumnsList),
+                Optional.of(newColumnsList), // The group key columns may be read unnecessarily, but they need to be tracked in the JdbcTableHandle
```

Comment:
"need ..." because? explain/hint at the "why".

Reply:
@alexjo2144 can you chime in. I assume you understand this part better.